core/arch/arm/plat-imx/drivers/imx_caam.c

```diff
@@ -53,6 +58,103 @@ static void caam_enable_clocks(bool enable)
 
 }
 
+static void imx_caam_reset_jr(struct imx_caam_ctrl *ctrl)
+{
+	uint32_t reg_val = 0;
+	uint32_t timeout = 1000;
+
+	io_write32((vaddr_t)&ctrl->jrcfg[0].jrcr, 1);
+	do {
+		reg_val = io_read32((vaddr_t)&ctrl->jrcfg[0].jrintr);
+		reg_val &= 0xc;
+	} while ((reg_val == 0x4) && --timeout);
```

Comment:
what is timeout elapses? panic?
If not, maybe timeout test is useless.

Reply:
I'm returning `TEE_ERROR_SECURITY` now.